symengine/basic.h

```diff
@@ -97,7 +97,15 @@ class Basic : public EnableRCPFromThis<Basic>
     mutable hash_t hash_; // This holds the hash value
 #endif // WITH_SYMENGINE_THREAD_SAFE
 public:
+#ifdef USE_VIRTUAL_TYPEID
     virtual TypeID get_type_code() const = 0;
+#else
+    TypeID type_code_;
+    inline TypeID get_type_code() const
+    {
+        return type_code_;
```

Comment:
Regarding the initializer lists, perhaps we can use something like this:

http://www.learncpp.com/cpp-tutorial/114-constructors-and-initialization-of-derived-classes/